unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Favorites/FavoriteButtonComponentView.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class FavoriteButtonComponentView : BaseComponentView, IComponentModelConfig<FavoriteButtonComponentModel>
+{
+    [Header("Prefab References")]
+    [SerializeField] private ButtonComponentView button;
+    [SerializeField] private Image buttonFill;
+    [SerializeField] private Color noFavoriteFillColor;
+    [SerializeField] private Color favoriteFillColor;
+
+    [Header("Configuration")]
+    [SerializeField] internal FavoriteButtonComponentModel model;
+
+    public event Action<string, bool> OnFavoriteChange;
+
+    private bool isFavorite;
+    private string placeUUID;
+
+    public override void Awake()
+    {
+        base.Awake();
+
+        button.onClick.RemoveAllListeners();
+        button.onClick.AddListener(SetFavorite);
+    }
+
+    public override void RefreshControl()
+    {
+        if (model == null)
+            return;
+
+        isFavorite = model.isFavorite;
+        SetFavorite();
+    }
+
+    private void SetFavorite()
+    {
+        isFavorite = !isFavorite;
+        if (isFavorite)
+        {
+            SetFavoriteVisuals();
+        }
+        else
+        {
+            SetNoFavoriteVisuals();
+        }
+
+        OnFavoriteChange?.Invoke(placeUUID, isFavorite);
+    }
+
+    private void SetNoFavoriteVisuals()
```

Comment:
Since it is a `private` method, I would rather collapse it to one function 
`
SetButtonVisuals(bool isFavorite) =>  buttonFill.color = isFavorite ? favoriteFillColor : noFavoriteFillColor;
`
that will also collapse 41-48 lines just to one call